src/FMData/ConnectionInfo.cs

```diff
@@ -21,5 +21,13 @@ public class ConnectionInfo
         /// Password to use when making the connection.
         /// </summary>
         public string Password { get; set; }
+        /// <summary>
+        /// AWS Cognito UserPoolID
+        /// </summary>
+        public string CognitoUserPoolID { get; set; } = "us-west-2_NqkuZcXQY";
+        /// <summary>
+        /// AWS Cognito ClientID
+        /// </summary>
+        public string CognitoClientID { get; set; } = "4l9rvl4mv5es1eep1qe97cautn";
     }
```

Comment:
I'm thinking these should not have any default values here, as they'll be specific to each user/system.

Reply:
I see the updated comment and link, I also found this link: https://help.claris.com/en/customer-console-help/content/create-fmid-token.html which points to the link in your commits. Makes sense, even though its not obvious at first blush.

Reply:
So I'll try to explain why I kept the default values (for now):
To authenticate with FileMaker Cloud, we have to authenticate using a Claris ID account and to do so, the first step would always be to query the FileMaker Cloud endpoint under: [json](https://www.ifmcloud.com/endpoint/userpool/2.2.0.my.claris.com.json)
My understanding is, that these "User pool" information's are "always" the same for everyone who is using Claris ID for external authentication.
So in that sense, it "should" make it easier for the user of the library:

- No lookup of the required values, they're already preconfigured
- No need to understand the AWS Cognito UserPool concept or even better, no need to know that AWS is used under the hood.

In the end, when a user wants to use the client with FileMaker Cloud, he'll use just another Uri and the FileMakerCloudAuthTokenProvider:

`conn.FmsUri = "https://ssp.account.filemaker-cloud.com";`
`var fm = new FileMakerRestClient(new HttpClient(), new FileMakerCloudAuthTokenProvider(conn));`

FileMakerCloudAuthTokenProvider ... I really should have used shorter names 😅.

BTW: Meanwhile I had time (sorry for the delay) to test the api and so far it's working pretty well. According to the statistics in the console I've already transferred 600 MB (limit is 120 GB) and so far I had only two "Timeout" related exceptions.

Reply:
Glad to hear the library is working well for you! I am hoping to merge this in and make a new v5 release soon. Obviously need to do a bit of other clean to include in the v5 breaking.

Reply:
It's working fantastic and it's simple to use!